pkg/server/node.go

```diff
@@ -1044,3 +1045,81 @@ func (n *Node) RangeFeed(
 	}
 	return nil
 }
+
+func (n *Node) UnsafeHealRange(
+	ctx context.Context, req *roachpb.UnsafeHealRangeRequest,
+) (_ *roachpb.UnsafeHealRangeResponse, rErr error) {
+	defer func() {
+		if r := recover(); r != nil {
+			rErr = errors.Errorf("%v", r)
+		}
+	}()
+
+	var expValue *roachpb.Value
+	var desc roachpb.RangeDescriptor
+	if err := n.storeCfg.DB.Txn(ctx, func(ctx context.Context, txn *client.Txn) error {
```

Comment:
The basic primitive should take the range descriptor. We can leave it to the higher level abstraction to give us an appropriate one. There will be two ways to get this range descriptor: either look through meta2 (like we're doing here), or broadcast something cluster-wide and see if there are any surviving replicas for the given range.